### PR TITLE
Add integration test for `rename_all` attribute

### PR DESCRIPTION
**Pull request by rsdy**
Simple test cases to ensure fields and enum variants are all renamed correctly using the `rename_all` macro attribute.

### CODE REVIEW

**Comment by rsdy** [comment]
Writing the test for `enum`s immediately popped out an error in the generated code. Will fix that here.

**Comment by rsdy** [comment]
The test for `enum`s also fixes the enum serialization code. Ready for review.

File: instant-xml-macros/src/de.rs [hunk 1]
```diff
@@ -11,22 +11,53 @@ pub(crate) fn from_xml(input: &syn::DeriveInput) -> TokenStream {
         Err(e) => return e.to_compile_error(),
     };
 
+    let name = match &meta.rename {
```
Comment (djc) [inline instant-xml-macros/src/de.rs:14]: I would like to split this refactoring off into a separate commit.

File: instant-xml-macros/src/ser.rs [hunk 2]
```diff
@@ -56,6 +61,8 @@ fn serialize_enum(
             ) -> Result<(), instant_xml::Error> {
 		serializer.write_str(match self { #variants })
             }
+
+            const KIND: ::instant_xml::Kind = ::instant_xml::Kind::Scalar;
```
Comment (djc) [inline instant-xml-macros/src/ser.rs:65]: This is arguably a separate change, which also merits its own commit.

File: instant-xml/tests/rename.rs [hunk 3]
```diff
@@ -23,7 +23,7 @@ fn renamed() {
 }
 
 #[test]
-fn rename_all() {
+fn rename_all_struct() {
```
Comment (djc) [inline instant-xml/tests/rename.rs:26]: Please squash this into the first commit.

File: instant-xml-macros/src/ser.rs (hunk 2 above)
Comment (rsdy) [reply instant-xml-macros/src/ser.rs:65]: This is part of fixing the serialization logic, though. This constant is part of the trait.

File: instant-xml-macros/src/de.rs (hunk 1 above)
Comment (rsdy) [reply instant-xml-macros/src/de.rs:14]: I can move `default_namespace` and `name` back to the `deserialize_struct` function, as ultimately they're not needed here, so the minimum changeset becomes cleaner.

As a separate PR it would probably make sense to split these into a struct and define `(de)?serialize_{enum,struct}` functions on that parameter struct. WDYT?

File: instant-xml-macros/src/ser.rs (hunk 2 above)
Comment (djc) [reply instant-xml-macros/src/ser.rs:65]: I guess it would be okay to have the required fixes in the same commit as the additional test(s).

**Review by djc — APPROVED** [review]
